lib/ddtrace/contrib/configuration/settings.rb

```diff
@@ -29,6 +29,16 @@ def [](name)
         def []=(name, value)
           respond_to?("#{name}=") ? send("#{name}=", value) : set_option(name, value)
         end
+
+        # Override set_option to allow service_prefix to be added.
+        def set_option(name, value)
+          if name.to_sym == :service_name && !value.nil?
+            tracer = get_option(:tracer)
+            super(name, "#{tracer.service_prefix}#{value}")
```

Comment:
`service_prefix` lives on the Tracer, despite not actually being used at all by the tracer. Then this configuration object is the only thing to actually reference and use it, creating a dependency that might not make sense. We're really just putting this on `Tracer` because `c.tracer` in the configuration API acts as a "general settings" right now: whatever isn't integration specific ends up there.

We might want to consider moving `service_prefix` to some kind of new "general settings" object, which could contain settings not actually applicable to the tracer or integrations themselves. It would help us avoid unnecessary entanglement between the tracer and the configuration parts of the code.